filesystem/squashfs_superblock.ksy

```diff
@@ -0,0 +1,54 @@
+meta:
+  id: squashfs_superblock
+  title: SquashFS superblock
+  file-extension:
+    - squashfs
+    - snap
+  endian: le
+  license: CC-0
+doc-ref: https://github.com/AgentD/squashfs-tools-ng/blob/master/doc/format.txt
+seq:
+  - id: superblock
+    type: superblock
+types:
+  superblock:
+    seq:
+      - id: magic
+        contents: 'hsqs'
+      - id: inode_count
+        type: u4
+      - id: mod_time
+        type: u4
+      - id: block_size
+        type: u4
+      - id: frag_count
+        type: u4
+      - id: compressor
+        type: u2
+      - id: block_log
```

Comment:
It'd be nice to add `doc` to each field non-fully described by its title. And also crosslink it to squashfs impls using `-orig-id`

Reply:
The field name `-orig-id` is not self-descriptive either. :D It is not in the .ksy spec. What is it for?

Reply:
First: we usually call a `field` an entry in `seq`.

I don't remember we have the established terminology for the stuff within `fields` and `instance`s, but I'd call it an attribute. Attrs beginning with `-` are not validated by the compiler against specs, but there are some conventions. `-orig-id` is used for names in original specs and source code. I.e. if in the original spec some field is called so, you can and should put it there. If in the reference impl a variable or a struct member to which the val is parsed is called so, you can also use it. Multiple names can be inserted using YAML arrays.

Reply:
Original name in spec.
```
 | u16  | block log     | The log2 of the block size. If the two fields do not|
 |      |               | agree, the archive is considered corrupted.         |
```

Added the `doc` , but https://ide.kaitai.io/ doesn't render anything specific for it. JS code also doesn't contain the comment.

Reply:
@AgentD I copied comments for the parser from your https://github.com/AgentD/squashfs-tools-ng/blob/master/doc/format.txt spec. It is okay for you that the parser with these comments is CC-0?

Reply:
I guess that it may make sense to add that fact into https://github.com/AgentD/squashfs-tools-ng/blob/master/COPYING.md , if he is OK with relicensing that doc.

Reply:
> @AgentD I copied comments for the parser from your https://github.com/AgentD/squashfs-tools-ng/blob/master/doc/format.txt spec. It is okay for you that the parser with these comments is CC-0?

I have no problem with that. In fact, one of the main reasons I wanted to document the SquashFS format was to enable others to create implementations without getting tangled up in a license mess (as reading the GPL'd squashfs-tools or kernel code as a starting point would).

Reply:
@AgentD, could you explicitly license the docs under a permissive license then, please?